### PR TITLE
Import v1.6.1

### PR DESCRIPTION
**Pull request by edenhill**

### CODE REVIEW

**Comment by nmyk** [comment]
👋 It appears as though merging #566 before this has caused the required version of `librdkafka` to be ahead of the bundled version.

```
# go get -v -t github.com/confluentinc/confluent-kafka-go/kafka
github.com/confluentinc/confluent-kafka-go/kafka/librdkafka_vendor
github.com/confluentinc/confluent-kafka-go/kafka
# github.com/confluentinc/confluent-kafka-go/kafka
internal-deps/src/github.com/confluentinc/confluent-kafka-go/kafka/00version.go:44:2: error: #error "confluent-kafka-go requires librdkafka v1.6.0 or later. Install the latest version of librdkafka from the Confluent repositories, see http://docs.confluent.io/current/installation.html"
   44 | #error "confluent-kafka-go requires librdkafka v1.6.0 or later. Install the latest version of librdkafka from the Confluent repositories, see http://docs.confluent.io/current/installation.html"
      |  ^~~~~
```

**Comment by edenhill** [comment]
Yeah, we recommend using go modules to pin to a specific version.

master/unversioned imports will be fixed as soon as this is merged.

File: kafka/api.html [hunk 1]
```diff
@@ -672,11 +677,26 @@ <h2 class="toggleButton" title="Click to hide Index section">
           func (c *Consumer) GetMetadata(topic *string, allTopics bool, timeoutMs int) (*Metadata, error)
          </a>
         </dd>
+        <dd>
+         <a href="#Consumer.GetRebalanceProtocol">
+          func (c *Consumer) GetRebalanceProtocol() string
+         </a>
```
Comment (mhowlett) [inline kafka/api.html:683]: not much point exposing this imo, and slightly better not to, but doesn't matter.

File: kafka/api.html [hunk 2]
```diff
@@ -2278,11 +2298,20 @@ <h3 id="Consumer.Assign">
     <pre>func (c *<a href="#Consumer">Consumer</a>) Assign(partitions []<a href="#TopicPartition">TopicPartition</a>) (err <a href="//golang.org/pkg/builtin/#error">error</a>)</pre>
     <p>
      Assign an atomic set of partitions to consume.
-This replaces the current assignment.
+    </p>
+    <p>
+     The .Offset field of each TopicPartition must either be set to an absolute
+starting offset (&gt;= 0), or one of the logical offsets (`kafka.OffsetEnd` etc),
+but should typically be set to `kafka.OffsetStored` to have the consumer
```
Comment (mhowlett) [inline kafka/api.html:2305]: In .net I use -1001 for this. it would have been better if that were not allowed..

**Review by mhowlett — APPROVED** [review]

**Pull request merged**